### DISCUSSION TITLE
TeamCityLogger with report metadata issues

### DISCUSSION THREAD

**Discussion by rubiasc** [nuke-build/nuke#1057] [Q&A] [answered]
### Usage Information
Nuke 6.2.1, .NET 6, TeamCity 2022.4.4, TeamCity.Dotnet.Integration 1.0.28

### Description
We use TeamCity's metadata reporting https://www.jetbrains.com/help/teamcity/reporting-test-metadata.html to add additional information to E2E Tests, like screenshots of browser state and so on which worked quite well so far.

After incorporating nuke into our build pipeline we see TeamCity's Build Log littered with the warning `Cannot find test [unnamed test] to add metadata /null`, albeit the metadata is attached successfully. This turns out to be quite an inconvenience and makes it harder to search for real problems in the log files.

Any tips/tricks/pointers on how to stop this warning from litterig the build log would be appreciated.

### Reproduction Steps
Any target executing DotNetTest with the TeamCity logger will do, for example

```
Target E2ETest => _ => _
  .DependsOn(Compile)
  .DependsOn(SetupDb)
  .Produces(ArtifactPath / "*.TestResults.trx => TestResults")
  .Executes(() =>
  {
    var testProjects = Solution.GetProjects($"*.E2ETests");
    testProjects.ForEach(project =>
      DotNetTest(cfg => cfg
        .SetProjectFile(project)
        .SetConfiguration(Configuration)
        .SetVerbosity(TestVerbosity)
        .EnableNoRestore()
        .EnableNoBuild()
        .When(TeamCity.Instance is not null, cfg => cfg.AddTeamCityLogger())
        .AddLoggers($"trx;LogFileName={ArtifactPath / $"{project.Name}.TestResults.trx"}")));
  });
```
Report Test-Metadata during or at the end of a test, this can be anything like a link to a screenshot for example 
```
Console.WriteLine ($"##teamcity[testMetadata type='image' value='TestReport.zip!/{screenshot.Name}']");
```

### Expected Behavior
No Warning, Metadata Information available in TeamCity's Test Information Tab

### Actual Behavior
`Cannot find test [unnamed test] to add metadata /null` warning for every test, Metadata Information available in TeamCity's Test Information Tab

### Known Workarounds
No workaround identified; trying to tinker with verbosity or report testMetadata with the full test name as specified by Jetbrains yielded the same results

**Comment by matkoch** [comment]
It seems like this is better posted/cross-linked to the TeamCity issue tracker? There's not really much that we do about this.

**Comment by rubiasc** [comment, marked as answer]
Sure thing, there we go https://youtrack.jetbrains.com/issue/TW-78679
